core/src/main/java/hudson/model/DirectoryBrowserSupport.java

```diff
@@ -404,12 +406,19 @@ private static void zip(OutputStream outputStream, VirtualFile dir, String glob)
          */
         private final boolean isReadable;
 
-        public Path(String href, String title, boolean isFolder, long size, boolean isReadable) {
```

Comment:
Public non-`@Restricted` constructor in non-`@Restricted` class was removed.

Reply:
It was replaced but in any case, I will revert this change.

Reply:
> replaced

That's the gist of the problem. Plugins compiled against core before this change is merged may call the old constructor and will not magically look for another one.

The canonical way to do this is to keep the old constructor around, deprecate it, and have it call the new one with a default value for the arg. Or just don't add the arg to the constructor, but go with setters instead. Either works.